optaplanner-core/src/main/java/org/optaplanner/core/api/score/stream/ConstraintCollectors.java

```diff
@@ -985,18 +991,24 @@ private static Runnable innerCountDistinctLong(CountDistinctLongResultContainer
     /**
      * Creates constraint collector that returns {@link SortedSet} of the same element type as the
      * {@link ConstraintStream}.
+     * <p>
+     * The contents of the collection will be changing during {@link org.optaplanner.core.api.solver.Solver} lifetime,
+     * they must therefore never be modified nor stored without making a defensive copy first.
```

Comment:
Can the users modify or store elements of that list? No, they can't.
In a vanilla CS, can the users modify or store the tuple facts? No, they can't.

I don't think this needs to be explicitly stated.